subprojects/gradle/namespaced-resources-fixer/src/main/kotlin/com/avito/android/plugin/FileResourcesFixer.kt

```diff
@@ -0,0 +1,172 @@
+package com.avito.android.plugin
+
+import io.github.detekt.parser.KtCompiler
+import org.jetbrains.kotlin.com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.com.intellij.psi.PsiWhiteSpace
+import org.jetbrains.kotlin.psi.KtDotQualifiedExpression
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.psi.KtImportDirective
+import org.jetbrains.kotlin.psi.KtImportList
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.psi.KtTreeVisitorVoid
+import java.io.File
+
+/**
+ * Replace unambiguous R class references to Application by libraries R classes
+ */
+internal class FileResourcesFixer {
+
+    private val compiler = KtCompiler()
+
+    /**
+     * @return new file content or null if nothing changed
+     */
+    fun fixMergedResources(file: File, app: AndroidModule, libraries: List<AndroidModule>): String? {
+        val ktFile = compiler.parse(file)
```

Comment:
Did you measure how long does it take to parse and fix all files in the project?

Reply:
About 40 seconds for the whole test source sets in the application module. 
The most expensive part is to build all modules.